### PR TITLE
BlackBoxBoundedContext improvements

### PR DESCRIPTION
**Pull request by dmdashenkov**
In this PR we provide a number of enhancements for the `BlackBoxBoundedContext` API:
 1) The Truth API is updated to version 1.0.
 2) Custom `Subject`s now expose the `actual()` method in order to allow users to build more complex verifications, for example, to test metadata of dispatched signals.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: server/src/test/java/io/spine/server/aggregate/AggregateTest.java [hunk 1]
```diff
@@ -867,14 +873,13 @@ void fromEventReact() {
...
-            BlackBoxBoundedContext
-                    .singleTenant()
-                    .with(new TaskAggregateRepository())
-                    .receivesCommand(reassignTask())
-                    .assertThat(acked(once()).withoutErrorsOrRejections())
-                    .assertThat(emittedEvents(AggCannotReassignUnassignedTask.class,
-                                              AggUserNotified.class))
-                    .close();
+            EventSubject assertEvents = context.receivesCommand(reassignTask())
+                                          .assertEvents();
```
Comment (armiol) [inline server/src/test/java/io/spine/server/aggregate/AggregateTest.java]: Looks the alignment is broken here.

File: testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java [hunk 2]
```diff
@@ -150,8 +145,46 @@ public final S withType(Class<? extends M> messageClass) {
...
+        FactKey(String value) {
+            this.value = value;
+        }
+
+        String value() {
+            return value;
         }
     }
 }
+
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java]: This line looks redundant to me.

File: testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java [hunk 3]
```diff
@@ -150,8 +145,46 @@ public final S withType(Class<? extends M> messageClass) {
...
+    /**
+     * Key strings for Truth facts.
+     */
+    @VisibleForTesting
+    enum FactKey {
+
+        MESSAGE_COUNT("the count of the generated messages is"),
+        REQUESTED_INDEX("but the requested index was"),
+        @SuppressWarnings("DuplicateStringLiteralInspection")
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java]: Please comment on this. Just for consistency.

File: testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java [hunk 4]
```diff
@@ -108,20 +98,25 @@ public final void isNotEmpty() {
      *
      * <p>Fails if the index is out of the range of the generated message sequence.
      */
-    public final ProtoSubject<?, ?> message(int index) {
-        int size = Iterables.size(actual());
-        if (index >= size) {
+    public final ProtoSubject message(int index) {
+        if (messages() == null) {
+            failWithActual(fact(FactKey.ACTUAL.value, null));
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java]: Can we use the enum values without referencing the enum name?

File: testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java [hunk 5]
```diff
@@ -108,20 +98,25 @@ public final void isNotEmpty() {
...
                                 .that(Empty.getDefaultInstance());
-        } else {
-            T outerObject = Iterables.get(actual(), index);
-            Message unpacked = AnyPacker.unpack(outerObject.getMessage());
-            return ProtoTruth.assertThat(unpacked);
         }
+        T outerObject = Iterables.get(messages(), index);
+        Message unpacked = AnyPacker.unpack(outerObject.getMessage());
+        return ProtoTruth.assertThat(unpacked);
+
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java]: Please kill this line.

File: testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java [hunk 6]
```diff
@@ -150,8 +145,46 @@ public final S withType(Class<? extends M> messageClass) {
...
+    private @Nullable Iterable<T> messages() {
+        return actual;
+    }
+
+    /**
+     * Obtains the list of messages under assertion.
+     *
+     * @return an immutable copy of the {@code actual} messages
+     */
+    public List<T> actual() {
```
Comment (armiol) [inline testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java]: I am not sure, but we tend to use the returning values of such methods as plain `List`s and then eventually wrap them into `ImmutableList`s over and over again. 

So if we can declare the `ImmutableList` as a type of the returning value, I would do so.

**Comment by dmdashenkov** [comment]
@armiol, PTAL again.

**Comment by armiol** [comment]
@dmdashenkov @dmitrykuzmin please settle down on the conflicts. We need `1.0.2` to release the Bootstrap in `1.0.2`.

**Pull request merged**